### PR TITLE
Fix rbox regularization

### PR DESCRIPTION
**Pull request by Y-T-G**
The description says it regularizes the angles to be between 0 and 90:
https://github.com/ultralytics/ultralytics/blob/bad9200cc00858532f90feef9eca0b191b7e79a7/ultralytics/utils/ops.py#L791-L793

But it isn't doing that. It is simply ensuring that width is always greater than height:
https://github.com/ultralytics/ultralytics/blob/bad9200cc00858532f90feef9eca0b191b7e79a7/ultralytics/utils/ops.py#L802

However that's not important for Ultralytics OBB because we're using the OpenCV definition (since we use `minAreaRect`), and not the LongEdge definition:
https://mmrotate.readthedocs.io/en/latest/intro.html
image

The important part for OpenCV definition is that the angle is always between 0 and 90.

Updated the function to regularize based on angles.

```python
from ultralytics import YOLO
import math
from ultralytics.engine.results import Results
from ultralytics.utils.ops import xywhr2xyxyxyxy, xyxyxyxy2xywhr , regularize_rboxes
import matplotlib.pyplot as plt
import torch
img = torch.ones(100,100,3, dtype=torch.uint8)
rbox = torch.tensor((50, 50, 15, 45, 245 * torch.pi / 180, 0.5, 0))[None]
print("Original:\t", rbox)
print("Regularize:\t", torch.cat((regularize_rboxes(rbox[:, :5]), rbox[:, 5:]), dim=-1))
print("Inverse Test:\t", (torch.cat((xyxyxyxy2xywhr(xywhr2xyxyxyxy(rbox[:, :5])), rbox[:, 5:]), dim=-1)))
results = Results(obb=rbox, orig_img=img.numpy(), path="", names={0:"0"})
plt.imshow(results.plot(conf=False, labels=False))
```

**main**
Regularized box doesn't match inverted box (`xywhr` to `xyxyxyxyxy` to `xywhr`).
```python
Original:        tensor([[50.0000, 50.0000, 15.0000, 45.0000,  4.2761,  0.5000,  0.0000]])
Regularize:      tensor([[50.0000, 50.0000, 45.0000, 15.0000,  2.7053,  0.5000,  0.0000]])
Inverse Test:    tensor([[50.0000, 50.0000, 15.0000, 45.0000,  1.1345,  0.5000,  0.0000]])
```

**PR**
Regularized box matches inverted box (`xywhr` to `xyxyxyxyxy` to `xywhr`).
```python
Original:        tensor([[50.0000, 50.0000, 15.0000, 45.0000,  4.2761,  0.5000,  0.0000]])
Regularize:      tensor([[50.0000, 50.0000, 15.0000, 45.0000,  1.1345,  0.5000,  0.0000]])
Inverse Test:    tensor([[50.0000, 50.0000, 15.0000, 45.0000,  1.1345,  0.5000,  0.0000]])
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
Improves the handling of rotated bounding boxes by refining their regularization logic. 🛠️✨  

### 📊 Key Changes  
- Adjusted logic in the `regularize_rboxes` function to swap width (`w`) and height (`h`) based on angle conditions, ensuring bounding box consistency.  
- Simplified angle regularization to handle values more efficiently, ensuring they stay within a valid range (mod `π/2`).  

### 🎯 Purpose & Impact  
- **Purpose**: Improves geometric consistency for rotated bounding boxes by fixing edge-case errors and ensuring better alignment. 🖼️  
- **Impact**: Users analyzing or processing rotated objects will see improved accuracy and reliability in their bounding box calculations, particularly for edge cases involving angles near `π/2`. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description. You’ve provided great detail here, but you may also want to link to any related issues in the [Issues tab](https://github.com/ultralytics/ultralytics/issues).
- ✅ **Synchronize with Source**: Confirm your PR is up to date with the `main` branch of the repository. If your branch is behind, update it using the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify that all [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. It's critical that any failing checks are resolved before merging.
- ✅ **Update Documentation**: If this change modifies functionality, please ensure any relevant [documentation](https://docs.ultralytics.com/) is updated to reflect these updates.
- ✅ **Add Tests**: Include or update tests where applicable to cover your changes, and verify that all test cases pass.
- ✅ **Sign the CLA**: If this is your first time contributing to Ultralytics, please ensure you’ve signed the [Contributor License Agreement (CLA)](https://docs.ultralytics.com/help/CLA/) by commenting "I have read the CLA Document and I sign the CLA" in this thread.
- ✅ **Provide an MRE**: If applicable, provide a **Minimum Reproducible Example (MRE)** to show how your change resolves the issue. You've already included both the main and PR examples—excellent! 🎉

For more information, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have further questions or need clarification, feel free to leave a comment, and we’ll address them as soon as possible. 

Thank you for contributing to Ultralytics! 🚀 An Ultralytics engineer will review your PR soon. 😊

**Comment by glenn-jocher** [comment]
@Y-T-G great explanation, thank you!

**Comment by UltralyticsAssistant** [comment]
🌟 Huge thanks to @Y-T-G and @glenn-jocher for this fantastic contribution! 🎉 Your work on refining the rotated bounding box logic is a game-changer that significantly enhances precision and consistency for edge cases. 🚀  

As Henry Ford once said, *"Quality means doing it right when no one is looking."* Your dedication to perfecting the small but critical details exemplifies this beautifully, and it will undoubtedly make a major impact for our users. 🖼️💡  

We’re proud to have such talented contributors—thank you for being part of the Ultralytics journey! ❤️

**Comment by Laughing-q** [comment]
@Y-T-G Hey thanks for the fix! 
I think in this PR when it deals with case `angle=pi/2` it'd regularize it to `angle=0`, is that what we want? or better we remain it as it is when `angle=pi/2` for consistency?
```python
from ultralytics import YOLO
import math
from ultralytics.engine.results import Results
from ultralytics.utils.ops import xywhr2xyxyxyxy, xyxyxyxy2xywhr , regularize_rboxes
import matplotlib.pyplot as plt
import torch
img = torch.ones(100,100,3, dtype=torch.uint8)
rbox = torch.tensor((50, 50, 15, 45, 90 * torch.pi / 180, 0.5, 0))[None]
print("Original:\t", rbox)
print("Regularize:\t", torch.cat((regularize_rboxes(rbox[:, :5]), rbox[:, 5:]), dim=-1))
print("Inverse Test:\t", (torch.cat((xyxyxyxy2xywhr(xywhr2xyxyxyxy(rbox[:, :5])), rbox[:, 5:]), dim=-1)))
results = Results(obb=rbox, orig_img=img.numpy(), path="", names={0:"0"})
plt.imshow(results.plot(conf=False, labels=False))
```
```bash
Original:        tensor([[50.0000, 50.0000, 15.0000, 45.0000,  1.5708,  0.5000,  0.0000]])
Regularize:      tensor([[50.0000, 50.0000, 15.0000, 45.0000,  0.0000,  0.5000,  0.0000]])
Inverse Test:    tensor([[50.0000, 50.0000, 15.0000, 45.0000,  1.5708,  0.5000,  0.0000]])
```

**Comment by Y-T-G** [comment]
BTW, this is a related issue highlighting this ambiguity:

https://github.com/ultralytics/ultralytics/issues/15771

**Comment by Laughing-q** [comment]
@Y-T-G yeah make sense. thanks!
